lib/msal-common/src/client/SPAClient.ts

```diff
@@ -318,18 +329,56 @@ export class SPAClient extends BaseClient {
         this.cacheManager.resetTempCacheItems(cachedState || "");
     }
 
+    /**
+     * Gets a valid authority object and resolves endpoints if needed.
+     * @param authorityUri 
+     */
+    private async getDiscoveredAuthority(authorityUri: string): Promise<Authority> {
```

Comment:
Maybe this along with `getAuthorityObject` belongs in Authority or AuthorityFactory?

Reply:
Not a bad idea, I will see if I can make that change.